qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -259,32 +286,36 @@ testlog "TEST: cloned images"
 testlog " - default"
 parent_image=test_parent
 parent_snap=snap
-create_image ${CLUSTER2} ${PARENT_POOL} ${parent_image}
+create_image_and_enable_mirror ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${RBD_MIRROR_MODE}
 write_image ${CLUSTER2} ${PARENT_POOL} ${parent_image} 100
 create_snapshot ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap}
 protect_snapshot ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap}
 
 clone_image=test_clone
 clone_image ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} ${clone_image}
 write_image ${CLUSTER2} ${POOL} ${clone_image} 100
-
-enable_mirror ${CLUSTER2} ${PARENT_POOL} ${parent_image} journal
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
+  enable_mirror ${CLUSTER2} ${POOL} ${clone_image} snapshot
+else
+  enable_mirror ${CLUSTER2} ${PARENT_POOL} ${parent_image} journal
+fi
 wait_for_image_replay_started ${CLUSTER1} ${PARENT_POOL} ${parent_image}
 wait_for_replay_complete ${CLUSTER1} ${CLUSTER2} ${PARENT_POOL} ${parent_image}
-wait_for_status_in_pool_dir ${CLUSTER1} ${PARENT_POOL} ${parent_image} 'up+replaying' 'primary_position'
+wait_for_replaying_status_in_pool_dir ${CLUSTER1} ${PARENT_POOL} ${parent_image}
 compare_images ${PARENT_POOL} ${parent_image}
 
 wait_for_image_replay_started ${CLUSTER1} ${POOL} ${clone_image}
 wait_for_replay_complete ${CLUSTER1} ${CLUSTER2} ${POOL} ${clone_image}
-wait_for_status_in_pool_dir ${CLUSTER1} ${POOL} ${clone_image} 'up+replaying' 'primary_position'
+wait_for_replaying_status_in_pool_dir ${CLUSTER1} ${POOL} ${clone_image}
 compare_images ${POOL} ${clone_image}
 remove_image_retry ${CLUSTER2} ${POOL} ${clone_image}
 
 testlog " - clone v1"
-clone_image ${CLUSTER1} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} ${clone_image}1
-
-clone_image ${CLUSTER2} ${PARENT_POOL} ${parent_image} ${parent_snap} ${POOL} \
-            ${clone_image}_v1 --rbd-default-clone-format 1
+clone_image_and_enable_mirror ${CLUSTER1} ${PARENT_POOL} \
+    ${parent_image} ${parent_snap} ${POOL} ${clone_image}1
+clone_image_and_enable_mirror ${CLUSTER2} ${PARENT_POOL} \
+    ${parent_image} ${parent_snap} ${POOL} ${clone_image}_v1 snapshot \
```

Comment:
Instead of hard coding "snapshot", I think it would be nicer to pass `RBD_MIRROR_MODE` like it's done for `create_image_and_enable_mirror()` -- again so that clone images are dealt with the same as standalone images.